src/py21cmfast/src/heating_helper_progs.c

```diff
@@ -836,8 +854,57 @@ double integrate_over_nu(double zp, double local_x_e, double lower_int_limit, in
 // table in the new version of 21CMMC (including spin-temperature fluctuations).
 
 typedef struct{
-    double nu_0, x_e, x_e_ave, ion_eff;
+    double nu_0, x_e, x_e_ave, ion_eff, ion_eff_MINI, log10_Mturn_MINI, LOG10_MTURN_INT;
 } tauX_params_approx;
+double tauX_integrand_approx_MINI(double zhat, void *params){
+
+    double n, drpropdz, nuhat, sigma_tilde, fcoll, HI_filling_factor_zhat;
+
+    int z_fcoll_int1,z_fcoll_int2;
+    float z_fcoll_val1,z_fcoll_val2;
+    // New in v1.4
+	double log10_Mturn_MINI, LOG10_MTURN_INT;
+	double fcoll_MINI_left, fcoll_MINI_right, fcoll_MINI;
+
+    int redshift_int_fcollz, log10_Mturn_MINI_int_fcollz;
+    float redshift_table_fcollz, log10_Mturn_MINI_table_fcollz;
+
+    tauX_params_approx *p = (tauX_params_approx *) params;
+
+    drpropdz = C * dtdz(zhat);
+    n = N_b0 * pow(1+zhat, 3);
+    nuhat = p->nu_0 * (1+zhat);
+	log10_Mturn_MINI = p->log10_Mturn_MINI;
+	LOG10_MTURN_INT = p->LOG10_MTURN_INT;
+
+    redshift_int_fcollz = (int)floor( ( zhat - determine_zpp_min )/zpp_bin_width );
+    redshift_table_fcollz = determine_zpp_min + zpp_bin_width*(float)redshift_int_fcollz;
+
+    fcoll = Nion_z_val[redshift_int_fcollz] + ( zhat - redshift_table_fcollz )*( Nion_z_val[redshift_int_fcollz+1] - Nion_z_val[redshift_int_fcollz] )/(zpp_bin_width);
+
+    log10_Mturn_MINI_int_fcollz = (int)floor( ( log10_Mturn_MINI - LOG10_MTURN_MIN) / LOG10_MTURN_INT);
```

Comment:
ah, now I understand the question. @andreimesinger do you remember why we have decided to do Fcoll(avg(logM_turn)) instead of avg(Fcoll((logM_turn))) for the ST normalization? I remember we had similar discussions.

Reply:
At high z, the PDF of logM_turn is very peaky (see the ipython notebook). So these two should be close.